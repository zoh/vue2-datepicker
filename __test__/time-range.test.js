import { mount } from '@vue/test-utils';
import TimeRange from '../src/time/time-range';

let wrapper;

afterEach(() => {
  wrapper.destroy();
});

describe('TimeRange', () => {
  it('render: correct classes of the columns', () => {
    wrapper = mount(TimeRange, {
      propsData: {
        format: 'hh:mm a',
        minuteStep: 30,
        hourStep: 2,
        value: [new Date(2019, 9, 4, 8, 30, 0), new Date(2019, 9, 4, 18, 30, 0)],
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('feat: change the end time when start > end', () => {
    wrapper = mount(TimeRange, {
      propsData: {
        value: [new Date(2019, 9, 4, 8, 30, 0), new Date(2019, 9, 4, 18, 30, 0)],
      },
    });
    const hour = wrapper.find('[data-type=hour] li:nth-child(20)');
    hour.trigger('click');
    expect(wrapper.emitted().select[0][0]).toEqual([
      new Date(2019, 9, 4, 19, 30, 0),
      new Date(2019, 9, 4, 19, 30, 0),
    ]);
  });

  it('supports defaultValue is Array', () => {
    wrapper = mount(TimeRange, {
      propsData: {
        defaultValue: [new Date(2019, 9, 1, 10), new Date(2019, 11, 1, 12)],
      },
    });
    const actived = wrapper.findAll('.active');
    expect(actived.at(0).text()).toBe('10');
    expect(actived.at(3).text()).toBe('12');
  });
});
