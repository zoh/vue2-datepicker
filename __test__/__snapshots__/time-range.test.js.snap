// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeRange render: correct classes of the columns 1`] = `
<div
  class="mx-range-wrapper"
>
  <div
    class="mx-time"
  >
    <!---->
     
    <div
      class="mx-time-content"
    >
      <div
        class="mx-time-columns"
      >
        <div
          class="mx-scrollbar mx-time-column"
          style="position: relative; overflow: hidden;"
        >
          <div
            class="mx-scrollbar-wrap"
            style="margin-right: -0px;"
          >
            <ul
              class="mx-time-list"
              data-index="0"
              data-type="hour"
            >
              <li
                class="mx-time-item"
                data-index="0"
              >
                
        12
      
              </li>
              <li
                class="mx-time-item"
                data-index="1"
              >
                
        02
      
              </li>
              <li
                class="mx-time-item"
                data-index="2"
              >
                
        04
      
              </li>
              <li
                class="mx-time-item"
                data-index="3"
              >
                
        06
      
              </li>
              <li
                class="mx-time-item active"
                data-index="4"
              >
                
        08
      
              </li>
              <li
                class="mx-time-item"
                data-index="5"
              >
                
        10
      
              </li>
            </ul>
          </div>
           
          <div
            class="mx-scrollbar-track"
          >
            <div
              class="mx-scrollbar-thumb"
            />
          </div>
        </div>
        <div
          class="mx-scrollbar mx-time-column"
          style="position: relative; overflow: hidden;"
        >
          <div
            class="mx-scrollbar-wrap"
            style="margin-right: -0px;"
          >
            <ul
              class="mx-time-list"
              data-index="1"
              data-type="minute"
            >
              <li
                class="mx-time-item"
                data-index="0"
              >
                
        00
      
              </li>
              <li
                class="mx-time-item active"
                data-index="1"
              >
                
        30
      
              </li>
            </ul>
          </div>
           
          <div
            class="mx-scrollbar-track"
          >
            <div
              class="mx-scrollbar-thumb"
            />
          </div>
        </div>
        <div
          class="mx-scrollbar mx-time-column"
          style="position: relative; overflow: hidden;"
        >
          <div
            class="mx-scrollbar-wrap"
            style="margin-right: -0px;"
          >
            <ul
              class="mx-time-list"
              data-index="2"
              data-type="ampm"
            >
              <li
                class="mx-time-item active"
                data-index="0"
              >
                
        AM
      
              </li>
              <li
                class="mx-time-item"
                data-index="1"
              >
                
        PM
      
              </li>
            </ul>
          </div>
           
          <div
            class="mx-scrollbar-track"
          >
            <div
              class="mx-scrollbar-thumb"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mx-time"
  >
    <!---->
     
    <div
      class="mx-time-content"
    >
      <div
        class="mx-time-columns"
      >
        <div
          class="mx-scrollbar mx-time-column"
          style="position: relative; overflow: hidden;"
        >
          <div
            class="mx-scrollbar-wrap"
            style="margin-right: -0px;"
          >
            <ul
              class="mx-time-list"
              data-index="0"
              data-type="hour"
            >
              <li
                class="mx-time-item"
                data-index="0"
              >
                
        12
      
              </li>
              <li
                class="mx-time-item"
                data-index="1"
              >
                
        02
      
              </li>
              <li
                class="mx-time-item"
                data-index="2"
              >
                
        04
      
              </li>
              <li
                class="mx-time-item active"
                data-index="3"
              >
                
        06
      
              </li>
              <li
                class="mx-time-item"
                data-index="4"
              >
                
        08
      
              </li>
              <li
                class="mx-time-item"
                data-index="5"
              >
                
        10
      
              </li>
            </ul>
          </div>
           
          <div
            class="mx-scrollbar-track"
          >
            <div
              class="mx-scrollbar-thumb"
            />
          </div>
        </div>
        <div
          class="mx-scrollbar mx-time-column"
          style="position: relative; overflow: hidden;"
        >
          <div
            class="mx-scrollbar-wrap"
            style="margin-right: -0px;"
          >
            <ul
              class="mx-time-list"
              data-index="1"
              data-type="minute"
            >
              <li
                class="mx-time-item"
                data-index="0"
              >
                
        00
      
              </li>
              <li
                class="mx-time-item active"
                data-index="1"
              >
                
        30
      
              </li>
            </ul>
          </div>
           
          <div
            class="mx-scrollbar-track"
          >
            <div
              class="mx-scrollbar-thumb"
            />
          </div>
        </div>
        <div
          class="mx-scrollbar mx-time-column"
          style="position: relative; overflow: hidden;"
        >
          <div
            class="mx-scrollbar-wrap"
            style="margin-right: -0px;"
          >
            <ul
              class="mx-time-list"
              data-index="2"
              data-type="ampm"
            >
              <li
                class="mx-time-item disabled"
                data-index="0"
              >
                
        AM
      
              </li>
              <li
                class="mx-time-item active"
                data-index="1"
              >
                
        PM
      
              </li>
            </ul>
          </div>
           
          <div
            class="mx-scrollbar-track"
          >
            <div
              class="mx-scrollbar-thumb"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
